internal/factsengine/gatherers/mountinfo_test.go

```diff
@@ -0,0 +1,212 @@
+	"testing"
+
+	"github.com/moby/sys/mountinfo"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	utilsMocks "github.com/trento-project/agent/pkg/utils/mocks"
+)
+
+type MountInfoTestSuite struct {
+	suite.Suite
+	mockMountParser *mocks.MountParserInterface
+	mockExecutor    *utilsMocks.CommandExecutor
+}
+
+func TestMountInfoTestSuite(t *testing.T) {
+	suite.Run(t, new(MountInfoTestSuite))
+}
+
+func (suite *MountInfoTestSuite) SetupTest() {
+	suite.mockMountParser = new(mocks.MountParserInterface)
+	suite.mockExecutor = new(utilsMocks.CommandExecutor)
+}
+
+func (suite *MountInfoTestSuite) TestMountInfoParsingSuccess() {
+	suite.mockMountParser.On("GetMounts", mock.Anything).Return([]*mountinfo.Info{
+		{
+			Mountpoint: "/sapmnt",
+			Source:     "10.1.1.10:/sapmnt",
+			FSType:     "nfs4",
+			Options:    "rw,relatime",
+		},
+		{
+			Mountpoint: "/hana/data",
+			Source:     "/dev/mapper/vg_hana-lv_data",
+			FSType:     "xfs",
+			Options:    "rw,relatime",
+		},
+		{
+			Mountpoint: "/hana/log",
+			Source:     "/dev/mapper/vg_hana-lv_log",
+			FSType:     "xfs",
+			Options:    "rw,relatime",
+		},
+	}, nil)
+
+	blkidOutput := []byte(`
+DEVNAME=/dev/mapper/vg_hana-lv_data
+UUID=82e3685c-6c2a-432b-bf8c-a71286248fae
+BLOCK_SIZE=4096
+TYPE=xfs
+`)
+
+	blkidOutputNoUUID := []byte(`
+DEVNAME=/dev/mapper/vg_hana-lv_log
+PARTUUID=82e3685c-6c2a-432b-bf8c-a71286248fae
+BLOCK_SIZE=4096
+TYPE=xfs
+`)
+
+	suite.mockExecutor.
+		On("Exec", "blkid", "10.1.1.10:/sapmnt", "-o", "export").
+		Return(nil, fmt.Errorf("blkid error")).
+		On("Exec", "blkid", "/dev/mapper/vg_hana-lv_data", "-o", "export").
+		Return(blkidOutput, nil).
+		On("Exec", "blkid", "/dev/mapper/vg_hana-lv_log", "-o", "export").
+		Return(blkidOutputNoUUID, nil)
+
+	requestedFacts := []entities.FactRequest{
+		{
+			Name:     "not_mounted",
+			Gatherer: "mount_info",
+			CheckID:  "check1",
+			Argument: "/usr/sap",
+		},
+		{
+			Name:     "shared",
+			Gatherer: "mount_info",
+			CheckID:  "check1",
+			Argument: "/sapmnt",
+		},
+		{
+			Name:     "mounted_locally",
+			Gatherer: "mount_info",
+			CheckID:  "check1",
+			Argument: "/hana/data",
+		},
+		{
+			Name:     "no_uuid",
+			Gatherer: "mount_info",
+			CheckID:  "check1",
+			Argument: "/hana/log",
+		},
+	}
+
+	gatherer := gatherers.NewMountInfoGatherer(suite.mockMountParser, suite.mockExecutor)
+
+	factResults, err := gatherer.Gather(requestedFacts)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:    "not_mounted",
+			CheckID: "check1",
+			Value: &entities.FactValueMap{
+				Value: map[string]entities.FactValue{
+					"block_uuid":  &entities.FactValueString{Value: ""},
+					"fs_type":     &entities.FactValueString{Value: ""},
+					"mount_point": &entities.FactValueString{Value: ""},
```

Comment:
For now, this is perfect IMHO, in the future if we want to also detect more complex scenarios we could decompose the mountpoint when it's not found and test backwards if the parent exists, e.g. `/usr/sap` passed as argument would first need to check `/usr/sap` and if doesn't find it, it would check `/usr` and if not found, it could then fallback to `/` and just return the details stating in `mount_point` the parent mountpoint of that path.

Reply:
Let's see if we have such a requirement hehe